src/IoHandlers.cpp

```diff
@@ -291,4 +291,67 @@ int IoEpoll::prepareNetwork()
 	return rc;
 }
 #endif
+#ifdef  USING_VMA_EXTRA_API
+//==============================================================================
+//------------------------------------------------------------------------------
+IoVmaPoll::IoVmaPoll(int _fd_min, int _fd_max, int _fd_num) : IoHandler(_fd_min, _fd_max,_fd_num, 0, 0){
+	m_current_vma_ring_comp = NULL;
+}
+
+//------------------------------------------------------------------------------
+IoVmaPoll::~IoVmaPoll() {
+  for (m_rings_vma_comps_map_itr = m_rings_vma_comps_map.begin(); m_rings_vma_comps_map_itr != m_rings_vma_comps_map.end(); ++m_rings_vma_comps_map_itr){
+  	FREE(m_rings_vma_comps_map_itr->second);
+  }
+}
+
+//------------------------------------------------------------------------------
+int IoVmaPoll::prepareNetwork()
+{
+	int rc = SOCKPERF_ERR_NONE;
+	int list_count = 0;
+	int ring_fd = 0;
+
+	printf("\n");
+	for (int ifd = m_fd_min; ifd <= m_fd_max; ifd++) {
+		if (g_fds_array[ifd]) {
+			ring_fd = 0;
+			g_vma_api->get_socket_rings_fds(ifd, &ring_fd,1);
+			assert((-1) != ring_fd);
```

Comment:
consider checking return value for `g_vma_api->get_socket_rings_fds(ifd, &ring_fd,1);` and correct formatting as `, 1);`